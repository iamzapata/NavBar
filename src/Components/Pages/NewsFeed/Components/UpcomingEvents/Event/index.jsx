import React from "react"
import { shape, string } from "prop-types"
import friendlyEventDate from "./helpers"
import "./Event.scss"

const Event = ({ event: { name, date, location } }) => {
  return (
    <div className="Event">
      <p className="Event__Name title is-size-6">{name}</p>
      <p className="subtitle is-size-7">
        <span className="Event__Date">{friendlyEventDate(date)}</span>
        {", "}
        <span className="Event__Location">{location}</span>
      </p>
    </div>
  )
}

Event.propTypes = {
  event: shape({
    name: string.isRequired,
    date: string.isRequired,
    location: string.isRequired
  }).isRequired
}

export default Event
