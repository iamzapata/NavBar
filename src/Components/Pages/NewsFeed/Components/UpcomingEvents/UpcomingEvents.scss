@import "styles/variables.scss";

.UpcomingEvents {
  padding: 1.5rem;

  span.is-size-7 {
    display: block;
    color: $color-light-gray-text;
    margin: 0 0 5px 0;
  }
}
