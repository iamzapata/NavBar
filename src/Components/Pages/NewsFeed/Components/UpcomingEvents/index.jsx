import React, { PureComponent } from "react"
import { arrayOf, shape, string, number } from "prop-types"
import { connect } from "react-redux"
import { doesArrayHaveData } from "utils/hasData"
import { eventsSelector } from "../../reducers/upcomingEvents"
import getUpcomingEvents from "../../actions/upcommingEvents"
import Event from "./Event"
import "./UpcomingEvents.scss"

class UpcomingEvents extends PureComponent {
  componentDidMount() {
    const { getUpcomingEvents } = this.props
    getUpcomingEvents()
  }
  render() {
    const { events } = this.props
    return (
      <div className="UpcomingEvents">
        <span className="is-size-7">UPCOMING EVENTS</span>
        {doesArrayHaveData(events) &&
          events.map(event => <Event key={event.id} event={event} />)}
      </div>
    )
  }
}

UpcomingEvents.propTypes = {
  events: arrayOf(
    shape({
      id: number,
      name: string,
      date: string,
      location: string
    })
  ).isRequired
}

const mapStateToProps = state => ({
  events: eventsSelector(state)
})

const mapDispatchToProps = {
  getUpcomingEvents
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UpcomingEvents)
