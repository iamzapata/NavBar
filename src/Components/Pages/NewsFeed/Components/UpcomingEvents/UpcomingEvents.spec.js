import React from "react"
import { render, cleanup } from "react-testing-library"
import { Provider } from "react-redux"
import ReduxThunk from "redux-thunk"
import configureMockStore from "redux-mock-store"
import UpcomingEvents from "./index"

const mockStore = configureMockStore([ReduxThunk])
const store = mockStore({
  Theme: { themeName: "Pinterest" },
  NewsFeed: { feed: [], isLoading: false },
  UserProfile: {
    profile: { bio: "", interests: "", first_name: "", last_name: "" },
    isLoading: false
  },
  UpcomingEvents: { events: [], isLoading: false }
})

describe("<UpcomingEvents />", () => {
  afterEach(cleanup)
  it("Should render self", () => {
    const { container } = render(
      <Provider store={store}>
        <UpcomingEvents />
      </Provider>
    )
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
