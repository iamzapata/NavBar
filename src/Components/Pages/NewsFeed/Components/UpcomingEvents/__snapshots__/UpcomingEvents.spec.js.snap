// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpcomingEvents /> Should render self 1`] = `
<div
  class="UpcomingEvents"
>
  <span
    class="is-size-7"
  >
    UPCOMING EVENTS
  </span>
</div>
`;
